### ISSUE TITLE
Spurious UP-TO-DATE false negatives

### ISSUE THREAD

**Issue by ypoluektovich** [etiennestuder/gradle-jooq-plugin#49]
I am seeing semi-reliable false negatives in up-to-dateness detection of the source generation task. Running gradle with `--info` shows this:

```
Executing task ':generateSampleJooqSchemaSource' (up-to-date check took 0.012 secs) due to:
  Value of input property 'configHash' has changed for task ':generateSampleJooqSchemaSource'
```

Needless to say, I am not actually changing the configuration between runs.

Output of `gradle --version`:

```
Gradle 3.5
------------------------------------------------------------

Build time:   2017-04-10 13:37:25 UTC
Revision:     b762622a185d59ce0cfc9cbc6ab5dd22469e18a6

Groovy:       2.4.10
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.8.0_131 (Oracle Corporation 25.131-b11)
OS:           Linux 4.11.3-1-ARCH amd64
```

**Comment by ypoluektovich** [comment]
It might also be important to note that I have gradle daemon turned off.

**Comment by C4Grey** [comment]
Same thing to me, I am using jOOQ along with Flybase migration:
```groovy
generateCreativeJooqSchemaSource.dependsOn flywayMigrate
compileKotlin.dependsOn generateCreativeJooqSchemaSource
```
Running my Spring Boot application like this: `gradle bootRun -Dorg.gradle.daemon=false`

**Comment by eric239** [comment]
same here; @etiennestuder any suggestions?

**Comment by eric239** [comment]
Bump... @etiennestuder if you still maintain this project would you please at least comment?

**Comment by etiennestuder** [comment]
Do you see the same behavior when the daemon is turned on?
Can you provide a reproducible example?

**Comment by eric239** [comment]
@etiennestuder it's not exactly trivial to build a self-contained example, specifically around the database part (I'm using MySQL; how do I "package" even a simple DB in?).

What I have confirmed (by tweaking the source to output `configHash` as it is computed in `JooqTask`):
1. **Without** daemon, the value does change between subsequent invocations (when nothing is changed in the project)
2. **With** daemon, the value does not change, and thus the code works as expected.

I had some issues in the past with the daemon, and I think it is fair to say that the plugin should work either way. Thus I have two questions for you:
1. Does the daemon thing offer any clues? I've never developed a gradle plugin myself so I really don't know why would it work differently
2. Can you perhaps consider an alternative way of computing the hash, e.g. serialize `Configuration` object and hash the result string? (Not sure if that would or wouldn't make a difference -- I guess it depends on the answer to the first question :-)

Thanks for looking into!

**Comment by ypoluektovich** [comment]
> Does the daemon thing offer any clues?

My guess is that restarting the daemon JVM changes the identity hash codes of objects somewhere within the configuration.

**Comment by eric239** [comment]
@ypoluektovich hence my suggestion to consider computing the hash off configuration serialized into string -- that should be consistent (in java string hashcode is deterministic)

**Comment by etiennestuder** [comment]
Thanks for the insights, @eric239. I will look into it.

If I can write out a String that is reproducible across non-daemon runs, I can also produce a hash code that is reproducible across builds ;-)

**Comment by etiennestuder** [comment]
Should be fixed by this PR: https://github.com/etiennestuder/gradle-jooq-plugin/pull/76

**Comment by etiennestuder** [comment]
Released in 3.0.1
